src/NuGet.Clients/NuGet.PackageManagement.UI/Models/ItemsChangeObservableCollectionBase.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+
+namespace NuGet.PackageManagement.UI
+{
+    public class ItemsChangeObservableCollection<T> : ObservableCollection<T>
+    {
+        public void Refresh()
+        {
+            OnCollectionChanged(new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
```

Comment:
Yeah it is a weird fix, but `ObservableCollection` only raises a notify when items are added and removed, in this particular case we are not doing that, thats why I needed to reset it whenever I wanted

Reply:
As mentioned, I think it's because `DisplayVersion` does not implement `INotifyPropertyChanged`. If you did that, you wouldn't need to create this class.

My concern is that people who are familiar with WPF and XAML already know about `ObservableCollection` and `INotifyPropertyChanged`. Therefore, when reading code, they can quickly read and understand assuming that everything works as expected. But `ItemsChangedObservableCollectionBase` is non-standard, so when reading code that uses it, they have to stop and think "how is this different to ObservableCollection?", and maybe go into the implemention to understand it.

Even if it takes longer to make `DisplayVersion` implement `INotifyPropertyChanged` than it was to create this extension of `ObservableCollection`, the end result might be more tech debt than if `DisplayVersion` was changed instead.

Reply:
Andy, we worked on adding a comment to this class (which was probably needed anyway). Let us know if that alleviates your concerns.